crates/bevy_pbr/src/light.rs

```diff
@@ -1,64 +1,46 @@
 use bevy_core::Byteable;
 use bevy_ecs::reflect::ReflectComponent;
 use bevy_reflect::Reflect;
-use bevy_render::{
-    camera::{CameraProjection, PerspectiveProjection},
-    color::Color,
-};
+use bevy_render::color::Color;
 use bevy_transform::components::GlobalTransform;
-use std::ops::Range;
 
 /// A point light
 #[derive(Debug, Reflect)]
 #[reflect(Component)]
-pub struct Light {
+pub struct PointLight {
     pub color: Color,
-    pub fov: f32,
-    pub depth: Range<f32>,
```

Comment:
I presume the original reasoning for why these extra fields were included, was to reuse the same `Light` type for both Point lights and (future) Spot lights?

I think I like this change. As a user, I think having multiple distinct clearly-named types for different kinds of lights is a good idea.

As for the implementation, I am not very familiar with how differently the different kinds of lights have to be handled for shading. I know Directional/Ambient are special, and those better be their own independent types anyway, but for Point/Spot I am not so sure.

If it's not going to complicate passing of light data and handling in the shader, then this change seems good, also from an implementation perspective. I wonder what the arguments for unifying them into a single type are?

Reply:
From experience I have a strong preference to not put in unused code like this 'because we'll use it soon'. It's simple enough to put it in the next PR and it's only confusing at this point.

For example, while working on shadowmaps, the above fields needed changing anyway...

Reply:
I think separate types for each light type makes sense, even if point vs spot have some technical overlap. I expect most of the changes in this pr to hold up even when we add spotlights.